### PR TITLE
City state election rigging

### PR DESCRIPTION
**Pull request by tuvus**
This PR continues #7610 by adding the city-state elections and election rigging. 
Previously I had a different style of election rigging that wasn't as close to what Civ V has. 

Now every city-state has a 15 turn election cycle (to be moved to a unique later) that is randomized at the start. Once this number hits 0 the city-state will hold an election and the spies that are present will try to rig it.

Note that spies cannot level up from rigging elections.

Coups are next to implement

<details><summary>Picture</summary>

SpyRigElection

</details>

### CODE REVIEW

File: core/src/com/unciv/logic/civilization/diplomacy/CityStateFunctions.kt [hunk 1]
```diff
@@ -60,11 +62,73 @@ class CityStateFunctions(val civInfo: Civilization) {
...
+            fun getVotesFromSpy(spy: Spy): Int {
+                var votes = (civInfo.getDiplomacyManager(spy.civInfo).influence / 2).toInt()
+                votes += (spy.getSkillModifier() * spy.getEfficiencyModifier()).toInt() // ranges from 30 to 90
+                return votes
+            }
+
+            val spiesVotes = spies.map { Pair(it, getVotesFromSpy(it)) }
+            var totalVotes = spiesVotes.sumOf { it.second } + 20
+
+            val randomSeed = capital.location.x * capital.location.y + 123f * civInfo.gameInfo.turns
```
[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/logic/civilization/diplomacy/CityStateFunctions.kt:91]: Yes, it will inform us once one gets merged in.

**Comment by tuvus** [comment]
I forgot to add that the rigging elections timer in the spy overview is a little wonky, it only updates at the end of the turn and ends up being a turn late.

File: core/src/com/unciv/logic/civilization/diplomacy/CityStateFunctions.kt [hunk 2]
```diff
@@ -60,11 +62,73 @@ class CityStateFunctions(val civInfo: Civilization) {
...
+            val spies = capital.espionage.getAllStationedSpies().filter { it.action == SpyAction.RiggingElections }
+            if (spies.isEmpty()) return
+
+            fun getVotesFromSpy(spy: Spy): Int {
+                var votes = (civInfo.getDiplomacyManager(spy.civInfo).influence / 2).toInt()
+                votes += (spy.getSkillModifier() * spy.getEfficiencyModifier()).toInt() // ranges from 30 to 90
+                return votes
+            }
+
+            val spiesVotes = spies.map { Pair(it, getVotesFromSpy(it)) }
```
[the remark this answers is not in the document]
Comment (tuvus) [reply core/src/com/unciv/logic/civilization/diplomacy/CityStateFunctions.kt]: It looks much cleaner with randomWeighted.

**Comment by yairm210** [comment]
Non-trivial conflicts

**Comment by tuvus** [comment]
Could we shorten the button Conducting Counter-Intelligence to Counter-Intelligence? It would buy us some more horizontal space on the espionage overview screen.

**Comment by yairm210** [comment]
Absolutely, you're the boss

**Review by yairm210 — APPROVED** [review]

**Pull request merged**